src/librustdoc/html/render/mod.rs

```diff
@@ -888,11 +910,20 @@ fn render_assoc_item(
             }
         };
         let vis = meth.visibility.print_with_space(meth.def_id, cx).to_string();
-        let constness = header.constness.print_with_space();
+        let mut constness = header.constness.print_with_space();
         let asyncness = header.asyncness.print_with_space();
         let unsafety = header.unsafety.print_with_space();
         let defaultness = print_default_space(meth.is_default());
         let abi = print_abi_with_space(header.abi).to_string();
+
+        if !constness.is_empty() {
+            if let Some(ConstStability { level: StabilityLevel::Unstable { .. }, .. }) =
+                meth.const_stability(cx.tcx())
+            {
+                constness = "";
+            }
+        }
```

Comment:
Why single this out for methods? I would expect this to be consistent wherever rustdoc prints `const`, including for functions. To enforce that you could pass in the const stability to `print_with_space` (by removing the trait implementation and just calling it directly).

Reply:
The `Constness` is defined in `rustc_hir`. I am going to add another trait just for `Constness` instead.

Reply:
Please do not add more traits, they make the code more complicated than necessary. A free function is fine.

Reply:
Okay, I will change that now.